qa-pipelines/flags.yml

```diff
@@ -4,37 +4,45 @@
 # The settings in this file will also be valid for our most complete pipeline
 # (which performs an upgrade and all tests)
 
-# deploy cf for upgrade pipelines
-enable-cf-deploy-pre-upgrade: true
+# Enable HA and non-HA pipelines
```

Comment:
Since we're modifying the way flags.yml is used, we need to update this section of the README: https://github.com/SUSE/cf-ci/tree/master/qa-pipelines#preset-file-instructions

Also, think setting the pre-upgrade cats flag to true here affects the functionality of the canonical pipeline, which can be deployed using cap-qa-full-upgrades.yml preset file, as mentioned in that same README section

Reply:
Oops, not sure when the symlink got deleted, but it shouldn't have been.

The flags file *is* a list of flags, with the default settings for the 'canonical' upgrade pipeline. As set-pipeline currently works, it still takes the list of all flags from that file, but sets any that aren't in the passed preset file to false. As it is, only 2 flags are false by default (the non-upgrade deploy shouldn't happen after an upgrade), because of the way concourse builds still had to run through all builds.

Honestly I think this was an incredibly hacky solution which was necessary at the time, but now that we're generating the pipeline manifest and only including the tasks which need to be run, I think we should move some of this logic into the generation script. Maybe we don't even need preset files anymore in their current form, though this could also part of a broader refactor that doesn't necessarily need to happen right now

Reply:
README updated in 5c05c5bcf031a6bfe03b100c61301f4f37a274b5.

I think preset files are good (at least, having configs on disk are good) because that makes it easier to reproduce pipelines. But we may want to consider doing different things later… but that's later.